pkg/scene/widget_align_system.go

```diff
@@ -0,0 +1,250 @@
+package scene
+
+import (
+	"github.com/reearth/reearth-backend/pkg/id"
+)
+
+// WidgetAlignSystem is the layout structure of any enabled widgets that will be displayed over the scene.
+type WidgetAlignSystem struct {
+	inner WidgetZone
+	outer WidgetZone
+}
+
+// WidgetZone is the structure of each layer (inner and outer) of the align system.
+type WidgetZone struct {
+	left   WidgetSection
+	center WidgetSection
+	right  WidgetSection
+}
+
+// WidgetSection is the structure of each section of the align system.
+type WidgetSection struct {
+	top    WidgetArea
+	middle WidgetArea
+	bottom WidgetArea
+}
+
+// WidgetArea has the widgets and alignment information found in each part area of a section.
+type WidgetArea struct {
+	widgetIds []*id.WidgetID
+	align     string
+}
+
+type WidgetLocation struct {
+	Zone    string
+	Section string
+	Area    string
+}
+
+const (
+	AlignStart  = "start"
+	AlignCenter = "center"
+	AlignEnd    = "end"
+)
+
+var Areas = []string{
+	"top",
+	"middle",
+	"bottom",
+}
+
+var Sections = map[string][]string{
+	"left":   Areas,
+	"center": Areas,
+	"right":  Areas,
+}
+
+var Zones = map[string]map[string][]string{
+	"inner": Sections,
+	"outer": Sections,
+}
+
+func (was *WidgetAlignSystem) FindWidgetIDLocation(wid id.WidgetID) (*int, *WidgetLocation) {
+	for z, s := range Zones {
+		for s2, a := range s {
+			for _, a2 := range a {
+				if i, h := was.Area(z, s2, a2).Has(wid); h {
+					wloc := WidgetLocation{z, s2, a2}
+					return i, &wloc
+				}
+			}
+		}
+	}
+	return nil, nil
+}
+
+// NewWidgetAlignSystem returns a new widget align system.
+func NewWidgetAlignSystem() *WidgetAlignSystem {
+	return &WidgetAlignSystem{}
+}
+
+// Zone will return a specific zone in the align system.
+func (was *WidgetAlignSystem) Zone(zone string) *WidgetZone {
+	if was == nil {
+		return nil
+	}
+	switch zone {
+	case "inner":
```

Comment:
```suggestion
	case scene.WidgetZoneInner:
```

constants are not used. Do the same for the others.